apps/src/templates/sectionProgress/UnitSelector.js

```diff
@@ -1,50 +1,82 @@
+import classNames from 'classnames';
 import PropTypes from 'prop-types';
-import React, {Component} from 'react';
-
-// TODO: Can/should we share any logic with AssignmentSelector?
-
-export const dropdownStyles = {
-  dropdown: {
-    display: 'block',
-    boxSizing: 'border-box',
-    fontSize: 'medium',
-    height: 34,
-    paddingLeft: 5,
-    paddingRight: 5,
-    width: 300,
-  },
-};
+import React from 'react';
+import {connect} from 'react-redux';
+
+import skeletonizeContent from '@cdo/apps/componentLibrary/skeletonize-content.module.scss';
+import {asyncLoadCoursesWithProgress} from '@cdo/apps/redux/unitSelectionRedux';
+
+import styles from './unit-selector.module.scss';
+
+function UnitSelector({
+  scriptId,
+  onChange,
+  coursesWithProgress,
+  asyncLoadCoursesWithProgress,
+  isLoadingCourses,
+}) {
+  React.useEffect(() => {
+    if (!coursesWithProgress || coursesWithProgress.length === 0) {
+      asyncLoadCoursesWithProgress();
+    }
+  }, [coursesWithProgress, asyncLoadCoursesWithProgress]);
+
+  const loadingSkeleton = () => (
+    <div>
+      <div
+        className={classNames(
+          styles.dropdown,
+          styles.skeletonDropdown,
+          skeletonizeContent.skeletonizeContent
+        )}
+      />
+    </div>
+  );
 
-export default class UnitSelector extends Component {
-  static propTypes = {
-    coursesWithProgress: PropTypes.array.isRequired,
-    scriptId: PropTypes.number,
-    onChange: PropTypes.func.isRequired,
-    style: PropTypes.object,
-  };
-
-  render() {
-    const {scriptId, onChange, coursesWithProgress} = this.props;
-
-    return (
-      <div>
-        <select
-          value={scriptId || undefined}
-          onChange={event => onChange(parseInt(event.target.value))}
-          style={dropdownStyles.dropdown}
-          id="uitest-course-dropdown"
-        >
-          {coursesWithProgress.map((version, index) => (
-            <optgroup key={index} label={version.display_name}>
-              {version.units.map(unit => (
-                <option key={unit.id} value={unit.id}>
-                  {unit.name}
-                </option>
-              ))}
-            </optgroup>
-          ))}
-        </select>
-      </div>
-    );
-  }
+  return isLoadingCourses ||
+    !coursesWithProgress ||
+    coursesWithProgress.length === 0 ? (
+    loadingSkeleton()
+  ) : (
+    <div>
+      <select
```

Comment:
nit: I see that changing this might be out of scope of this task, but I wonder if we can use SimpleDropdown here to slightly improve ui? Wasn't able to find this dropdown locally so not sure if DSCO SimpleDropdown is applicable here in terms of styles/visual appearance